api/types/client.go

```diff
@@ -285,10 +285,19 @@ type ServiceCreateOptions struct {
 }
 
 // ServiceCreateResponse contains the information returned to a client
-// on the  creation of a new service.
+// on the creation of a new service.
 type ServiceCreateResponse struct {
 	// ID is the ID of the created service.
 	ID string
+	// Warning is a non-fatal warning message to pass on to the user.
+	Warning string `json:",omitempty"`
```

Comment:
I don't think another field is necessary in addition to a `string` Warning field, particularly because we're not taking any specific action if the pinning by digest fails.